src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private static final double COEFFICIENT = 1.5;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.size = 0;
+        elements = (T[]) new Object[CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            grow();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkAddRange(index);
+        if (size == elements.length) {
+            grow();
+        }
+        System.arraycopy(elements, index,
+                elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkRange(index);
+        return elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkRange(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkRange(index);
+        T resulArray = (T) elements[index];
```

Comment:
There is a typo in the variable name 'resulArray'. It should be 'resultArray' for better readability and to avoid confusion.